daemon/delete_test.go

```diff
@@ -34,9 +35,9 @@ func TestContainerDoubleDelete(t *testing.T) {
 	// Mark the container as having a delete in progress
 	container.SetRemovalInProgress()
 
-	// Try to remove the container when its start is removalInProgress.
-	// It should ignore the container and not return an error.
-	if err := daemon.ContainerRm(container.ID, &types.ContainerRmConfig{ForceRemove: true}); err != nil {
-		t.Fatal(err)
+	// Try to remove the container when its state is removalInProgress.
+	// It should return an error indicating it is under removal progress.
+	if err := daemon.ContainerRm(container.ID, &types.ContainerRmConfig{ForceRemove: true}); err == nil {
+		t.Fatalf("expected err: %v, got nil", "removal of container %s is already in progress", container.ID)
```

Comment:
sorry for my mistake. Thanks for you help. :) updated that. Actually so sleepy that I have to go to bed.  See you tomorrow, my friend.